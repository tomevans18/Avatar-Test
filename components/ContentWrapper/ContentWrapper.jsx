import React from 'react';
import PropTypes from 'prop-types';

import Content from './ContentWrapper.style';

const propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
};

const ContentWrapper = ({ children, ...otherProps }) => (
  <Content {...otherProps}>{children}</Content>
);

ContentWrapper.propTypes = propTypes;

export default ContentWrapper;
