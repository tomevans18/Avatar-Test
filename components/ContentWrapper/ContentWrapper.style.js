import styled, { css } from 'styled-components';

export default styled.div`
  ${p =>
    p.flex &&
    css`
      display: flex;
      flex-direction: row;
    `} position: relative;
  width: 100%;
  max-width: 67rem;
  margin: 0 auto;
  box-sizing: border-box;
  padding: 0 calc(env(safe-area-inset-right) + ${p => p.theme.component.spacing});

  @supports (padding: max(0px)) {
    padding: 0 max(${p => p.theme.component.spacing}, env(safe-area-inset-left));
  }
`;
