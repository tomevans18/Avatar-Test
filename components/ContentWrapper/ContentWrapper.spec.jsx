import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ContentWrapper from './ContentWrapper';

describe('Content Wrapper', () => {
  it('should render as expected', () => {
    const wrapper = shallow(
      <ContentWrapper>
        <div>TEST ContentWrapper</div>
      </ContentWrapper>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render as expected with flex layout', () => {
    const wrapper = shallow(
      <ContentWrapper flex>
        <div>TEST ContentWrapper</div>
      </ContentWrapper>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
