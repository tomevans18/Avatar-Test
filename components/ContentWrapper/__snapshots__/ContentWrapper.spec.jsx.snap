// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Wrapper should render as expected 1`] = `
<styled.div>
  <div>
    TEST ContentWrapper
  </div>
</styled.div>
`;

exports[`Content Wrapper should render as expected with flex layout 1`] = `
<styled.div
  flex={true}
>
  <div>
    TEST ContentWrapper
  </div>
</styled.div>
`;
